src/hazelcore/semantics/UHExp.re

```diff
@@ -340,3 +340,65 @@ let text_operand =
       u_gen,
     );
   };
+
+let rec is_complete_line = (l: line, check_type_holes: bool): bool => {
+  switch (l) {
+  | EmptyLine => true
+  | LetLine(pat, _, body) =>
```

Comment:
this should check the type annotation if check_type_holes is true

Reply:
fixed